src/sage/matrix/matrix_polynomial_dense.pyx

```diff
@@ -4079,3 +4084,462 @@ cdef class Matrix_polynomial_dense(Matrix_generic_dense):
                 if P[j, j].degree() + shifts[j] <= degree_bound:
                     column_indices.append(j)
             return P[:,column_indices]
+
+    def _basis_completion_via_reversed_approx(self):
+        r"""
+        Return a Smith form-preserving nonsingular completion of a row basis of
+        this matrix. For a more detailed description, see
+        :meth:`basis_completion`, in the row-wise case.
+
+        EXAMPLES::
+
+            sage: pR.<x> = GF(7)[]
+        """
+        from sage.matrix.constructor import matrix # for identity
+
+        ring,var = self.base_ring().objgen()
+        m = self.nrows()
+        n = self.ncols()
+
+        # corner cases: after this, m>0 and n>0
+        if m == 0 :
```

Comment:
I think you should remove the space before `:` here (and everywhere else in your ticket).

Reply:
Done, thanks!